events/interactionCreate.js

```diff
@@ -1,8 +1,28 @@
+const {commandRoles} = require('../config/slashCommandPermissions');
+
+function isAuthorized(interaction, commandRoles) {
+  const allowedRoles = commandRoles.get(interaction.commandName);
+  if (
+    allowedRoles[0] === '@everyone' ||
+    interaction.member.roles.cache.some((r) => allowedRoles.includes(r.id))
+  ) {
+    return true;
+  }
+  return false;
+}
+
 module.exports = {
   name: 'interactionCreate',
   async execute(interaction) {
     if (!interaction.isCommand()) return;
 
+    if (!isAuthorized(interaction, commandRoles)) {
+      return await interaction.reply({
+        content: `You don't have permissions for this command`,
```

Comment:
Can change `You don't have permissions for this command` to `You don't have permission to use this command.`.

Reply:
Done